lib/realtime_web/channels/realtime_channel.ex

```diff
@@ -356,29 +355,39 @@ defmodule RealtimeWeb.RealtimeChannel do
   def handle_in(
         "access_token",
         %{"access_token" => refresh_token},
-        %{assigns: %{pg_sub_ref: pg_sub_ref, pg_change_params: pg_change_params}} = socket
+        %{
+          assigns: %{
+            access_token: access_token,
+            pg_sub_ref: pg_sub_ref,
+            pg_change_params: pg_change_params
+          }
+        } = socket
       )
       when is_binary(refresh_token) do
     socket = assign_counter(socket) |> assign(:access_token, refresh_token)
 
     case confirm_token(socket) do
       {:ok, claims, confirm_token_ref} ->
-        cancel_timer(pg_sub_ref)
+        if refresh_token == access_token do
```

Comment:
I think this would be much more clear if you put it in another function head in a guard.

Like: `when is_binary(access_token) and access_token == refresh_token`

Also this needs a `cancel_timer`.